### PR TITLE
gh #109 code clean up for audio

### PR DESCRIPTION
**Pull request by ishthiyaqahmed**
code clean up for the audio module.

### CODE REVIEW

File: src/test_l1_dsAudio.c [hunk 1]
```diff
@@ -10045,54 +8741,42 @@ void test_l1_dsAudio_negative_dsAudioAtmosCapsChangeRegisterCB(void) {
...
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(handle, null_handle);
 
-	        // Step 03: supported ARC types of the connected ARC/eARC
-                result = dsGetSupportedARCTypes(*(handle+i), &types);
+                // Step 03: supported ARC types of the connected ARC/eARC
+                result = dsGetSupportedARCTypes(handle, &types);
                 UT_ASSERT_EQUAL(result, dsERR_NONE);
-             } else {
+            } else {
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8769]: code never reaches else part, as the ports are filtered at 8758. check source type outside the for loop. check for other arc function

**Review by bhanucbp — COMMENTED** [review]
address the review comments

File: src/test_l1_dsAudio.c [hunk 2]
```diff
@@ -1423,69 +1271,58 @@ void test_l1_dsAudio_positive_dsSetDolbyVolumeMode(void) {
...
+         UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
+    } else {
+         UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
+    }
 
-        // Step 05: Terminate audio ports
-        result = dsAudioPortTerm();
+    // Step 04: Get the port handle for all supported audio ports
+    for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+        result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, &handle);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:1326]: done

File: src/test_l1_dsAudio.c (hunk 1 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 3]
```diff
@@ -6913,52 +6027,41 @@ void test_l1_dsAudio_negative_dsEnableAudioPort(void) {
  * 
  */
 void test_l1_dsAudio_positive_dsEnableMS12Config(void) {
-	// Start of the test
-	gTestID = 87; // Assuming Test Case ID is 87
-	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    // Start of the test
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:5943]: done

File: src/test_l1_dsAudio.c [hunk 4]
```diff
@@ -4751,49 +4168,38 @@ void test_l1_dsAudio_negative_dsGetStereoMode(void) {
...
-                UT_ASSERT_EQUAL(result, dsERR_NONE);
-                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
-
-                // Step 03: Set the stereo mode
-                for(int j = 0; j < gDSAudioPortConfiguration[i].no_of_supported_stereo_mode; j++)
-                {
-                        result = dsSetStereoMode(*(handle+i), gDSAudioPortConfiguration[i].supported_stereo_mode[j]);
-                        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Step 03: Set the stereo mode
+        for(int j = 0; j < gDSAudioPortConfiguration[i].no_of_supported_stereo_mode; j++)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 5]
```diff
@@ -4437,56 +3897,43 @@ void test_l1_dsAudio_negative_dsGetMS12AudioProfileList(void) {
...
-                if((gSourceType == 0) && (gDSAudioPortConfiguration[i].typeid == dsAUDIOPORT_TYPE_SPEAKER)) {
-                        // Step 03: Get the current MS12 audio profile for Speaker. MS12 Audio Profile is supported only on SPEAKER Port
-                        result = dsGetMS12AudioProfile(*(handle+i), currentProfile);
-                        UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-                        // Step 04: Verify that the profile string is valid and not empty
-                        UT_ASSERT_NOT_EQUAL(strlen(currentProfile), 0);
-                }
+        if((gSourceType == 0) && (gDSAudioPortConfiguration[i].typeid == dsAUDIOPORT_TYPE_SPEAKER)) {
+            // Step 03: Get the current MS12 audio profile for Speaker. MS12 Audio Profile is supported only on SPEAKER Port
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:3836]: done

File: src/test_l1_dsAudio.c [hunk 6]
```diff
@@ -4259,67 +3742,55 @@ void test_l1_dsAudio_negative_dsSetGraphicEqualizerMode(void) {
...
+        // Step 03: Get the list of supported MS12 audio profiles for each port
+        result = dsGetMS12AudioProfileList(handle, &profileList1);
+        if(gSourceType == 0) {
+            UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-                // Step 04: Verify that the profiles list is valid and not empty
-                UT_ASSERT_EQUAL(profileList1.audioProfileCount, 0);
-                UT_ASSERT_EQUAL(strlen(profileList1.audioProfileList), 0);
+            // Step 04: Verify that the profiles list is valid and not empty
+            UT_ASSERT_EQUAL(profileList1.audioProfileCount, 0);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 7]
```diff
@@ -3824,69 +3362,58 @@ void test_l1_dsAudio_positive_dsSetMISteering(void) {
...
+         UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
+    } else {
+         UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
+    }
 
-        // Step 05: Terminate audio ports
-        result = dsAudioPortTerm();
+    // Step 04: Get the port handle for all supported audio ports
+    for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+        result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, &handle);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 8]
```diff
@@ -2008,63 +1778,52 @@ void test_l1_dsAudio_negative_dsGetVolumeLeveller(void) {
...
+        {
+            // Step 03: Set Volume Leveller with valid mode and level for each audio port
+            // Assuming dsVolumeLeveller_t is a struct or similar with mode and level fields
+            for (int mode = 0; mode <= 2; mode++) {
+                for (int level = 0; level <= 10; level++) {
+                    volLeveller.mode = mode;
+                    volLeveller.level = level;
+                    result = dsSetVolumeLeveller(handle, volLeveller);
+                    UT_ASSERT_EQUAL(result, dsERR_NONE);
+                    } // end of level for loop
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 9]
```diff
@@ -1674,54 +1489,43 @@ void test_l1_dsAudio_negative_dsGetIntelligentEqualizerMode(void) {
...
-                } else {
-                        int mode = 5;
-                        result = dsSetIntelligentEqualizerMode(*(handle+i), mode);
-                        UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
-                }
+        if(gDSAudioPortConfiguration[i].ms12_capabilites & 0x02)
+        {
+        // Step 03: Set Intelligent Equalizer Mode for each port with a valid mode (0 to 6)
+            for (int mode = 0; mode <= 6; mode++) {
+            result = dsSetIntelligentEqualizerMode(handle, mode);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 10]
```diff
@@ -950,58 +866,47 @@ void test_l1_dsAudio_positive_dsGetDialogEnhancement(void) {
...
-        // allocate memory for the handles
-        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
-        if(handle == NULL)
-        {
-                UT_LOG_ERROR("Failed to allcoate memory for handlers");
-                return;
-        }
+    // Step 01: Attempt to get dialog enhancement without initializing
+    result = dsGetDialogEnhancement(-1, &dialogEnhancementLevel);
+    CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 11]
```diff
@@ -1026,62 +931,49 @@ void test_l1_dsAudio_negative_dsGetDialogEnhancement(void) {
...
-                        result = dsSetDialogEnhancement(*(handle+i), max_de_level);
-                        UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
-
-                }
-        }
-
-        // Step 06: Terminate audio ports
-        result = dsAudioPortTerm();
-        UT_ASSERT_EQUAL(result, dsERR_NONE);
+                    result = dsSetDialogEnhancement(handle, max_de_level);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 12]
```diff
@@ -297,53 +297,42 @@ void test_l1_dsAudio_negative_dsAudioPortTerm (void)
...
+           result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index ,&handle);
+           UT_ASSERT_EQUAL(result, dsERR_NONE);
+           UT_ASSERT_NOT_EQUAL(handle, null_handle);
 
-                // Remember last handle for comparison in next step
-                if (i == (gDSAudioNumberOfPorts-1)) {
-                        lastHandle = *(handle+i);
-                }
+           // Remember last handle for comparison in next step
+        if (i == (gDSAudioNumberOfPorts-1)) {
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 13]
```diff
@@ -10213,55 +8795,45 @@ void test_l1_dsAudio_negative_dsGetSupportedARCTypes(void) {
...
+                    UT_ASSERT_NOT_EQUAL(handle, null_handle);
 
-	// Step 04: Terminate audio ports
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+                    // Step 03: Set SAD for HDMI ARC port
+                    result = dsAudioSetSAD(handle, sadlist);
+           }        UT_ASSERT_EQUAL(result, dsERR_NONE);
+    } else {
+            UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8827]: comparison is wrong, call setSAD and then compare the result

**Review by bhanucbp — COMMENTED** [review]
address review comment

File: src/test_l1_dsAudio.c (hunk 13 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:8828]: done

**Review by bhanucbp — APPROVED** [review]
ship it

**Review by akhilbhas — APPROVED** [review]
Looks good

**Pull request merged**